gen/enum.go

```diff
@@ -110,19 +109,27 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 					return nil
 			<end ->
 				default:
-					<$val>, err := <$strconv>.ParseInt(<$s>, 10, 32)
+					<if checkEnumTextMarshalStrict ->
+						return <$fmt>.Errorf("unknown enum value %q for %q", <$s>, "<$enumName>")
+					<else ->
+						<$val>, err := <import "strconv">.ParseInt(<$s>, 10, 32)
```

Comment:
Why do we need to break UnmarshalText? Doesn't it make sense to be liberal about what we accept here?

And if we do need to break it, why only UnmarshalText and not UnmarshalJSON which also accepts numbers?